packages/editor/src/components/block-styles/index.js

```diff
@@ -68,13 +70,25 @@ function BlockStyles( {
 	onChangeClassName,
 	name,
 	attributes,
+	type,
 	onSwitch = noop,
 	onHoverClassName = noop,
 } ) {
 	if ( ! styles || styles.length === 0 ) {
 		return null;
 	}
 
+	if ( ! type.styles && ! find( styles, 'isDefault' ) ) {
```

Comment:
Is this `! type.styles ` really necessary?

Reply:
This condition checks whether the block type originally contains styles, whereas `styles` contains _all_ styles (original styles + ones added through `registerBlockStyle`).

The default style only needs to be added when the block type didn't already contain styles in the beginning.

So yeah, it's necessary :-)

Reply:
> The default style only needs to be added when the block type didn't already contain styles in the beginning.

I'm not certain I understand why? Why can't we always add a default style anytime we're missing one?

Reply:
Right. Yeah I guess I did some wrong thinking there 🤔